pkg/goldpinger/client.go

```diff
@@ -54,6 +55,68 @@ func CheckNeighboursNeighbours(ctx context.Context) *models.CheckAllResults {
 	return CheckAllPods(ctx, SelectPods())
 }
 
+// CheckCluster does a CheckNeighboursNeighbours and analyses results to produce a binary OK or not OK
+func CheckCluster(ctx context.Context) *models.ClusterHealthResults {
+
+	start := time.Now()
+	response := models.ClusterHealthResults{
+		GeneratedAt: strfmt.DateTime(start),
+	}
+	selectedPods := SelectPods()
+
+	// precompute the expected set of nodes
+	expectedNodes := []string{}
+	for _, peer := range selectedPods {
+		expectedNodes = append(expectedNodes, peer.HostIP)
+	}
+	sort.Strings(expectedNodes)
+
+	// get the response we serve for check_all
+	checkAll := CheckAllPods(ctx, selectedPods)
+
+	// 1. check that all nodes report OK
+	for _, resp := range checkAll.Responses {
+		if *resp.OK {
+			response.NodesHealthy = append(response.NodesHealthy, resp.HostIP.String())
+		} else {
+			response.NodesUnhealthy = append(response.NodesUnhealthy, resp.HostIP.String())
+			response.OK = false
+		}
+		response.NodesTotal++
+	}
+	// 2. check that all nodes report the expected peers
+	if len(checkAll.Responses) < 1 {
+		response.OK = false
+	} else {
+		for _, resp := range checkAll.Responses {
```

Comment:
Wondering if it might be better to add this check within the first loop instead of using two different loops?

Reply:
Should we also check DNS results?

Reply:
This should already be included in the endpoints' OK response.